Sources/firebase/include/FirebaseFirestore.hh

```diff
@@ -0,0 +1,42 @@
+
+#ifndef firebase_include_FirebaseFirestore_hh
+#define firebase_include_FirebaseFirestore_hh
+
+#include <firebase/firestore.h>
+
+// Functions defined in this namespace are used to get around the lack of
+// virtual function support currently in Swift. As that support changes
+// these functions will go away whenever possible.
+namespace swift_firebase::swift_cxx_shims::firebase::firestore {
+
+inline ::firebase::firestore::DocumentReference
+firestore_document(const ::firebase::firestore::Firestore firestore, ::std::string document_path) {
+  return firestore.Document(
+    document_path
+  );
+}
+
+/**
+ * @brief Get the Firestore instance from a given Document
+ *
+ * @param[in] document The document from which the Firestore instance should be extracted from.
+ *
+ * @return A pointer to the Firestore instance the passed in Document is in.
+*/
```

Comment:
Hmm, I don't mind the doc comments, but I think being consistent about adding or not adding them is better.

Reply:
Yeah I'll probably just remove this entirely since there's no point as it's literally just calling the underlying Firebase impl

Reply:
Done